workflows/diagnostics/fv3net/diagnostics/offline/compute.py

```diff
@@ -250,16 +222,77 @@ def _get_data_mapper_if_exists(config):
 
 
 def _variables_to_load(model):
-    return list(
+    vars = list(
         set(list(model.input_variables) + list(model.output_variables) + [DELP])
     )
+    if "Q2" in model.output_variables:
+        vars.append("water_vapor_path")
+    return vars
 
 
 def _add_derived_diagnostics(ds):
     merged = xr.merge([ds, derived_registry.compute(ds, n_jobs=1)])
     return merged.assign_attrs(ds.attrs)
 
 
+def _coarsen_transform(evaluation_resolution: int, prediction_resolution: int):
+    coarsening_factor = prediction_resolution // evaluation_resolution
+    logger.info(
+        f"Making predictions at validation data's c{prediction_resolution} resolution."
+    )
+    logger.info(f"Evaluating diagnostics at c{evaluation_resolution} resolution.")
+    if prediction_resolution % evaluation_resolution != 0:
+        raise ValueError(
+            "Evaluation grid resolution does not match or evenly divide prediction "
+            "resolution."
+        )
+    if coarsening_factor > 1:
+        prediction_grid = load_grid_info(f"c{prediction_resolution}")
+        logger.info(f"Coarsening predictions by factor of {coarsening_factor}.")
+        return coarsen_cell_centered(
+            weights=prediction_grid.area, coarsening_factor=coarsening_factor
+        )
+
+
+def get_prediction(
```

Comment:
Could this function do some logging? When testing this locally with C384 data the console goes silent for awhile (I think here).